src/main/java/seedu/toluist/model/Task.java

```diff
@@ -61,4 +61,22 @@ public void replaceTags(ArrayList<Tag> tags) {
     public boolean isOverdue() {
         return startDateTime != null && startDateTime.isBefore(LocalDateTime.now());
     }
+
+    public boolean isStringContainedInDescriptionIgnoreCase(String comparison) {
+        return this.description.toLowerCase().contains(comparison.toLowerCase());
+    }
+
+    public boolean isStringContainedInAnyTagIgnoreCase(String comparison) {
+        for (Tag tag : allTags) {
+            if (tag.tagName.toLowerCase().contains(comparison.toLowerCase())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public int compareTo(Task comparison) {
```

Comment:
But if we have to pass in a custom comparator everytime (which is likely to happen considering tthe differentt sorting use case), this `compareTo` will turn out to be useless. Comparable is more for things where the precedence order is consistent, like numbers or strings